tests/test_agent.py

```diff
@@ -769,3 +770,17 @@ def foo():
     assert mod.my_agent.name is None
     assert mod.foo() == snapshot('success (no tool calls)')
     assert mod.my_agent.name == 'my_agent'
+
+
+async def test_model_settings_override() -> None:
+    my_agent = Agent('test', model_settings={'temperature': 0.5})
```

Comment:
better to use a `FunctionModel` and return settings as JSON than introducing patch here.

Reply:
I think this also might not be relevant if we don't change https://github.com/pydantic/pydantic-ai/pull/227#discussion_r1882996108

I'm not opposed to doing both, but I think valuable to use a patch here to confirm that the merging is working as expected for all other models that use this pattern for passing `model_settings`